Assets/Project/Modules/Enemies/Turret/Scripts/scripts/TurretMediator.cs

```diff
@@ -0,0 +1,67 @@
+using System.Collections;
+using System.Collections.Generic;
+using Popeye.Core.Pool;
+using Popeye.Core.Services.ServiceLocator;
+using Popeye.Modules.CombatSystem;
+using Popeye.Modules.Enemies.Components;
+using Popeye.Modules.VFX.ParticleFactories;
+using UnityEngine;
+
+namespace Popeye.Modules.Enemies
+{
+    public class TurretMediator : AEnemyMediator
+    {
+        [Header("COMPONENTS")] 
+        [SerializeField] private TurretShooting _turretShooting;
+
+        public Transform playerTransform { get; private set; }
+        
+        private Core.Pool.ObjectPool _projectilePool;
+        private TurretMindEnemy _turretMind;
+        [SerializeField] private ParabolicProjectile _parabolicProjectile;
+        internal override void Init()
+        {
+            _projectilePool = new ObjectPool(_parabolicProjectile);
+            _projectilePool.Init(15);
```

Comment:
We will move this code into a Factory in the next tasks. 
But it is still incorrect, because the "new" and "init()" should be called only once, and the TurretMEdiator init() is likely to be called more than once, because it will get recycled.

These 2 lines should be in the Start() method